test/Ocelot.AcceptanceTests/Steps.cs

```diff
@@ -73,29 +54,41 @@ public class Steps : IDisposable
         private HttpContent _postContent;
         private BearerToken _token;
         public string RequestIdKey = "OcRequestId";
-        private readonly Random _random;
+        private readonly Random _random = new();
+        private readonly string _ocelotConfigFileName = $"{Guid.NewGuid():N}-ocelot.json";
         private IWebHostBuilder _webHostBuilder;
         private WebHostBuilder _ocelotBuilder;
         private IWebHost _ocelotHost;
         private IOcelotConfigurationChangeTokenSource _changeToken;
-
-        public Steps()
-        {
-            _random = new Random();
-        }
+        private bool _disposedValue;
 
         public async Task ThenConfigShouldBe(FileConfiguration fileConfig)
         {
             var internalConfigCreator = _ocelotServer.Host.Services.GetService<IInternalConfigurationCreator>();
-
             var internalConfigRepo = _ocelotServer.Host.Services.GetService<IInternalConfigurationRepository>();
-            var internalConfig = internalConfigRepo.Get();
 
+            var internalConfig = internalConfigRepo.Get();
             var config = await internalConfigCreator.Create(fileConfig);
 
             internalConfig.Data.RequestId.ShouldBe(config.Data.RequestId);
         }
 
+        public async Task ThenConfigShouldBeWithTimeout(FileConfiguration fileConfig, int timeoutMs)
+        {
+            var result = await Wait.WaitFor(timeoutMs).Until(async () =>
+            {
+                var internalConfigCreator = _ocelotServer.Host.Services.GetService<IInternalConfigurationCreator>();
+                var internalConfigRepo = _ocelotServer.Host.Services.GetService<IInternalConfigurationRepository>();
+
+                var internalConfig = internalConfigRepo.Get();
```

Comment:
Can we extract it to upper context, because it is initial config, right? And initial config can be read outside of waiting task, right?

Reply:
Can we just keep it like that for the minute, We could create a new PR later, and implement a better design for Steps.

Reply:
Please include this changes while working on #1706 if you wish